app/facades/board_game_facade.rb

```diff
@@ -3,19 +3,26 @@
 class BoardGamesFacade
   def self.search(name)
     data = BoardGamesService.find_board_games(name)
-
     available_categories = BoardGamesFacade.category
+
     data.map do |game|
-      categories = []
-      game[:categories].each do |game_category|
-        available_categories.each do |category|
-          if game_category[:id] == category.id
-            categories << category.name
-          end
+      categories = BoardGamesFacade.translate_ids(game, available_categories)
+
+      BoardGame.new(game, categories)
+    end
+  end
+
+  def self.translate_ids(game, available_categories)
+    translated_categories = []
+
+    game[:categories].each do |game_category|
+      available_categories.each do |category|
+        if game_category[:id] == category.id
+          translated_categories << category.name
```

Comment:
Simply abstracted this away from the creation of boardgames.

Reply:
Looks good!